teachertool/src/components/DebouncedTextarea.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef } from "react";
+import { Textarea, TextareaProps } from "react-common/components/controls/Textarea";
+
+export interface DebouncedTextareaProps extends TextareaProps {
+    intervalMs?: number; // Default 500 ms
+}
+
+// This functions like the React Common Input, but debounces onChange calls,
+// so if onChange is called multiple times in quick succession, it will only
+// be executed once after a pause of the specified `interval` in milliseconds.
+export const DebouncedTextarea: React.FC<DebouncedTextareaProps> = ({ intervalMs = 500, ...props }) => {
```

Comment:
I believe textarea has a prop to enable a resize handler on the corner of the control. If so, it would be great to enable that here:
![image](https://github.com/microsoft/pxt/assets/12176807/e4aeabea-e504-4755-aa0f-186097901139)

Reply:
Here are the props for textarea https://github.com/microsoft/pxt/blob/697fe51d4c563df3f78508f75fb8eb2a8c99934d/react-common/components/controls/Textarea.tsx#L4-L20 I have `resize` set to vertical, but it doesn't really seem to do anything.. Maybe it's because I've set a `max-height`?

Reply:
Yeah, I was thinking that, too @thsparks, and I thought the resize field would do that, but it didn't grow as more text was input. Maybe it has something to do with `cols`?

Reply:
Hmm, did you explore doing it through css rather than props? There may be a way through that.

Reply:
No, I hadn't... I'll look into it.

Reply:
It looks like this is going to be tricky-ish. Definitely more work that shouldn't be added to this PR, I'll make an item for it on our board. 

@eanders-ms, in regards to the resize question, it looks like we can change set it, but it looks really funky and is not functioning in the way that I would expect, so I'm going to leave it out. I agree with Thomas, though, that I think the better approach here would just have this auto-resize. I think it would be cool for users to be able to change the size of the textarea for a bit more customization, but again, I think it can be something I look into at a later time.

Reply:
Auto resize would be a fine solution. Let's track it as followup work.